cmd/conduit/root/root.go

```diff
@@ -0,0 +1,262 @@
+	// Map database configuration
+	c.flags.DBType = c.cfg.DB.Type
+	c.flags.DBPostgresConnectionString = c.cfg.DB.Postgres.ConnectionString
+	c.flags.DBPostgresTable = c.cfg.DB.Postgres.Table
+	c.flags.DBSQLiteTable = c.cfg.DB.SQLite.Table
+
+	// Map API configuration
+	c.flags.APIEnabled = c.cfg.API.Enabled
+	c.flags.APIHTTPAddress = c.cfg.API.HTTP.Address
+	c.flags.APIGRPCAddress = c.cfg.API.GRPC.Address
+
+	// Map logging configuration
+	c.flags.LogLevel = c.cfg.Log.Level
+	c.flags.LogFormat = c.cfg.Log.Format
+
+	// Map pipeline configuration
+	c.flags.PipelinesExitOnDegraded = c.cfg.Pipelines.ExitOnDegraded
+	c.flags.PipelinesErrorRecoveryMinDelay = c.cfg.Pipelines.ErrorRecovery.MinDelay
+	c.flags.PipelinesErrorRecoveryMaxDelay = c.cfg.Pipelines.ErrorRecovery.MaxDelay
+	c.flags.PipelinesErrorRecoveryBackoffFactor = c.cfg.Pipelines.ErrorRecovery.BackoffFactor
+	c.flags.PipelinesErrorRecoveryMaxRetries = c.cfg.Pipelines.ErrorRecovery.MaxRetries
+	c.flags.PipelinesErrorRecoveryMaxRetriesWindow = c.cfg.Pipelines.ErrorRecovery.MaxRetriesWindow
+
+	// Map schema registry configuration
+	c.flags.SchemaRegistryType = c.cfg.SchemaRegistry.Type
+	c.flags.SchemaRegistryConfluentConnectionString = c.cfg.SchemaRegistry.Confluent.ConnectionString
+
+	// Map preview features
+	c.flags.PreviewPipelineArchV2 = c.cfg.Preview.PipelineArchV2
+
+	// Map development profiling
+	c.flags.DevCPUProfile = c.cfg.Dev.CPUProfile
+	c.flags.DevMemProfile = c.cfg.Dev.MemProfile
+	c.flags.DevBlockProfile = c.cfg.Dev.BlockProfile
+
+	// Update paths
+	c.flags.DBSQLitePath = c.cfg.DB.SQLite.Path
+	c.flags.DBBadgerPath = c.cfg.DB.Badger.Path
+	c.flags.PipelinesPath = c.cfg.Pipelines.Path
+	c.flags.ConnectorsPath = c.cfg.Connectors.Path
+	c.flags.ProcessorsPath = c.cfg.Processors.Path
+}
+
+func (c *RootCommand) Execute(_ context.Context) error {
+	if c.flags.Version {
+		_, _ = fmt.Fprintf(os.Stdout, "%s\n", conduit.Version(true))
+		return nil
+	}
+
+	// 1. Load conduit configuration file and update general config.
+	if err := internal.LoadConfigFromFile(c.flags.ConduitConfigPath, &c.cfg); err != nil {
+		return err
+	}
+
+	// 2. Load environment variables and update general config.
+	if err := internal.LoadConfigFromEnv(&c.cfg); err != nil {
+		return err
+	}
+
+	// 3. Update the general config from flags.
+	c.updateConfigFromFlags()
+
+	// 4. Update flags from global configuration (this will be needed for conduit init)
+	c.updateFlagValuesFromConfig()
+
+	e := &conduit.Entrypoint{}
+	e.Serve(c.cfg)
+	return nil
+}
+
+func (c *RootCommand) Usage() string { return "conduit" }
+func (c *RootCommand) Flags() []ecdysis.Flag {
+	flags := ecdysis.BuildFlags(&c.flags)
+
+	currentPath, err := os.Getwd()
+	if err != nil {
+		panic(cerrors.Errorf("failed to get current working directory: %w", err))
+	}
+	c.cfg = conduit.DefaultConfigWithBasePath(currentPath)
+
+	conduitConfigPath := filepath.Join(currentPath, "conduit.yaml")
+	flags.SetDefault("config.path", conduitConfigPath)
```

Comment:
This might be a thing that we could try adding to ecdysis, completely mapping a struct to a flag set. We're already taking some info from the tags, we could probably also take the values and the field names from the config struct too.